inspirehep/modules/workflows/workflows/edit_article.py

```diff
@@ -41,6 +47,24 @@ def update_record(obj, eng):
     record.update(obj.data)
     record.commit()
 
+    user_id = obj.id_user
+    if user_id:
+        _set_transaction_user_id_for_last_record_update(control_number, obj.id_user)
+
+    db.session.commit()
+
+
+def _set_transaction_user_id_for_last_record_update(control_number, user_id):
+    record = get_db_record('lit', control_number)
+    revision = record.revisions[record.revision_id]
```

Comment:
I'll still put it inside
```python
try:
except IndexError:
```
and log if the revision not found, and just for readability will move the function before ``update_record``